src/Akeneo/Bundle/ElasticsearchBundle/Cursor/Cursor.php

```diff
@@ -0,0 +1,167 @@
+     * @param int                           $pageSize
+     */
+    public function __construct(
+        Client $esClient,
+        CursorableRepositoryInterface $repository,
+        array $esQuery,
+        $indexType,
+        $pageSize
+    ) {
+        $this->repository = $repository;
+        $this->esClient = $esClient;
+        $this->esQuery = $esQuery;
+        $this->indexType = $indexType;
+        $this->pageSize = $pageSize;
+
+        $this->items = $this->getNextItems($esQuery);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function current()
+    {
+        return current($this->items);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function next()
+    {
+        $currentItem = $this->current();
+
+        if (false === next($this->items)) {
+            $searchAfterIdentifier = $currentItem->getIdentifier();
+            $this->items = $this->getNextItems($this->esQuery, $searchAfterIdentifier);
+            $this->rewind();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function key()
+    {
+        return key($this->items);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function valid()
+    {
+        return !empty($this->items);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rewind()
+    {
+        reset($this->items);
+    }
+
+    /**
+     * Get the next items (hydrated from doctrine repository)
+     *
+     * @param array       $esQuery
+     * @param string|null $searchAfter
+     *
+     * @return array
+     */
+    private function getNextItems(array $esQuery, $searchAfter = null)
+    {
+        $identifiers = $this->getNextIdentifiers($esQuery, $searchAfter);
+        if (empty($identifiers)) {
+            return [];
+        }
+
+        return $this->repository->getItemsFromIdentifiers($identifiers);
+    }
+
+    /**
+     * Get the next identifiers from elasticsearch query
+     *
+     * @param array       $esQuery
+     * @param string|null $searchAfter
+     *
+     * @return array
+     *
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-search-after.html
+     */
+    private function getNextIdentifiers(array $esQuery, $searchAfter = null)
+    {
+        $esQuery['size'] = $this->pageSize;
+
+        if (!isset($esQuery['sort'])) {
+            $esQuery['sort'] = ['_uid' => 'asc'];
+        }
+
+        if (null !== $searchAfter) {
+            $esQuery['search_after'] = [$this->indexType . '#' . $searchAfter];
+        }
+
+        $response = $this->esClient->search($this->indexType, $esQuery);
+        $this->count = $response['hits']['total'];
+
+        $identifiers = [];
+        foreach ($response['hits']['hits'] as $hit) {
+            $identifiers[] = $hit['_source']['identifier'];
+        }
+
+        return $identifiers;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return $this->count;
```

Comment:
Does this count should return count of all the results? Because it's not clear on the ES documentation ( https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-search-after.html ). If the search after is like a query, it may return the count of results only /after/ the params. But in term of syntax, it's not part of  a query, so maybe ES implements it properly already (real results count).
Do we have integration tests on this point?